packages/core/src/StateMachine.ts

```diff
@@ -90,7 +94,13 @@ export class StateMachine<
     const resolvedContext =
       typeof context === 'function'
         ? context({
-            spawn: createSpawner(actorCtx?.self, this, actions),
+            spawn: createSpawner(
+              actorCtx?.self,
+              this,
+              undefined as any, // TODO: this requires `| undefined` for all referenced inputs that are spawnable in the context factory,
```

Comment:
```suggestion
              undefined as any, // TODO: this requires `| undefined` for all referenced dynamic inputs that are spawnable in the context factory,
```
This is the only problematic part of this PR from my PoV. Alternatively, we could only allow input-less actors to be spawned within the context factory... but that's perhaps even worse. A good thing is that you can always use `spawn('child', { input })` in the context factory.